src/transaction/components/TransactionProvider.test.tsx

```diff
@@ -440,6 +440,38 @@ describe('TransactionProvider', () => {
       );
     });
   });
+
+  it('should call setLifeCycleStatus when calling fallbackToWriteContract when user rejects request', async () => {
+    const writeContractsAsyncMock = vi
+      .fn()
+      .mockRejectedValue(new Error(METHOD_NOT_SUPPORTED_ERROR_SUBSTRING));
+    const writeContractAsyncMock = vi
+      .fn()
+      .mockRejectedValue({ cause: { name: 'UserRejectedRequestError' } });
+    (useWriteContracts as ReturnType<typeof vi.fn>).mockReturnValue({
+      status: 'idle',
+      writeContractsAsync: writeContractsAsyncMock,
+    });
+    (useWriteContract as ReturnType<typeof vi.fn>).mockReturnValue({
+      status: 'IDLE',
```

Comment:
idle or IDLE?

I have a feeling it's always `lowercase`

Reply:
yeah you're right it is `idle`